File: WebDriverTests_Java/WebDriverTests/src/test/java/core/systemControls/Link.java
package core.systemControls;


import org.openqa.selenium.By;

public class Link extends HtmlControl
{

    public Link(By locator)
    {
        this(locator, null);
    }

    public Link(By locator, HtmlControl container)
    {
        super(locator, container);
    }

    @Override
    public boolean getIsEnabled()
    {
        return !hasClass("disabled");
    }

    public final String getHref()
    {
        return getAttributeValue("href");
    }


    public static Link ByLinkText(String linkText)
    {
        return ByLinkText(linkText, null);
    }

    public static Link ByLinkText(String linkText, HtmlControl container)
    {
        return new Link(By.linkText(linkText), container);
    }
}

File: WebDriverTests_Csharp/Autotest.Core/Autotests.Utilities/WebTestCore/SystemControls/Link.cs
using OpenQA.Selenium;

namespace Autotests.Utilities.WebTestCore.SystemControls
{
    public class Link : HtmlControl
    {
        public Link(By locator, HtmlControl container = null)
            : base(locator, container)
        {
        }

        public Link(string idLocator, HtmlControl container = null)
            : base(idLocator, container)
        {
        }

        public override bool IsEnabled
        {
            get { return !HasClass("disabled"); }
        }

        public string Href
        {
            get { return GetAttributeValue("href"); }
        }

        public static Link ByLinkText(string linkText, HtmlControl container = null)
        {
            return new Link(By.LinkText(linkText), container);
        }
    }
}